### DISCUSSION TITLE
Inline math rendered as display math

### DISCUSSION THREAD

**Discussion by nico778** [remarkjs/remark#1343] [Q&A]
In my nextjs project i am pulling mdx files from a separate github repo and rendering them.
This works pretty well, except I can't figure out how to use inline math.
In github everything looks as expected, but when I run my nextjs project every inline math block is treated as display math.
What am I doing wrong? Sorry if this is something very basic, but I searched quite a while for a solution but could not find anything.

contentlayer config file:
```ts
export default makeSource((sourceKey = 'main') => ({
  syncFiles: (contentDir) =>
    syncContentFromGit({ contentDir, gitTag: sourceKey }),
  contentDirPath: `blog-${sourceKey}`,
  contentDirInclude: [BLOG_DIRECTORY, "public/images"],
  documentTypes: [Post],
  disableImportAliasWarning: true,
  mdx: {
    remarkPlugins: [
      remarkMath, 
    ],
    rehypePlugins: [
      rehypeMathJax, {
        tex: {
          inlineMath: [['$', '$']],
        },
      },
      rehypePrettyCode, {
        theme: 'one-dark-pro'
      }
    ]            
  },
}));
```

**Comment by wooorm** [comment]
Hey!

Info missing (no markdown posted) so hard to help you.

> What am I doing wrong? Sorry if this is something very basic, but I searched quite a while for a solution but could not find anything.

Presumably, as follows:
There is no standard for math in markdown.
How GH added math in markdown is buggy. I’d go as far as say **very bad**.
`remark-math` does not attempt to match GH. It instead matches exactly how code works, but with `$`.
Which code is “inline” or “block” is thus equal to which math is “inline” or “block”.

Please see the readme of `remark-math`: https://github.com/remarkjs/remark-math/tree/main/packages/remark-math#use. It has an example. And a “Syntax” heading, linking to more info on the syntax.

**Comment by nico778** [comment]
Thank you for the answer, the issue is that for me GH does work well, but my own site does not.

mdx file:
`#### (a) Analysis of Quantum States
For the given quantum states, determine their validity and calculate the probabilities of observing the states $|0\rangle$ and $|1\rangle$ when measured in the standard computational basis.`

On Github:
![gh](https://github.com/remarkjs/remark/assets/36104041/ddfcf00a-8c38-4005-9306-16792eff02fa)

On my own nextjs site:
![next](https://github.com/remarkjs/remark/assets/36104041/b198b038-0a2a-461a-a4ca-70c6727c7876)

**Reply by wooorm** [reply]
I think you are talking about how the CSS that GH has is different from the CSS you have.
The solution is: change your CSS.

**Comment by nico778** [comment]
ah okay, that makes sense. Thanks for pointing me in the right direction.
